### ISSUE TITLE
setTimeout followed by clearTimeout cause the first test to fail

### ISSUE THREAD

**Issue by lideming** [denoland/deno#7844]
File `test.ts`:
```
clearTimeout(setTimeout(() => { }, 1000));

Deno.test("noop test 1", () => { });
Deno.test("noop test 2", () => { });
Deno.test("noop test 3", () => { });
```

Run `deno test test.ts`:
```
Check file:///.../$deno$test.ts
running 3 tests
test noop test 1 ... FAILED (2ms)
test noop test 2 ... ok (1ms)
test noop test 3 ... ok (1ms)

failures:

noop test 1
AssertionError: Test case is leaking async ops.
Before:
  - dispatched: 1
  - completed: 0
After:
  - dispatched: 2
  - completed: 2

Make sure to await all promises returned from Deno APIs before
finishing test case.
    at assert (deno:cli/rt/06_util.js:33:13)
    at asyncOpSanitizer (deno:cli/rt/40_testing.js:44:7)
    at async Object.resourceSanitizer [as fn] (deno:cli/rt/40_testing.js:68:7)
    at async TestRunner.[Symbol.asyncIterator] (deno:cli/rt/40_testing.js:240:13)
    at async Object.runTests (deno:cli/rt/40_testing.js:317:22)

failures:

        noop test 1

test result: FAILED. 2 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out (5ms)

```

Version: Deno v1.4.4
OS: Windows 10 v2004 x64

**Comment by lideming** [comment]
There is a workaround, await something after clearTimeout, before running the test:
```ts
Deno.test("noop test 1", () => { });
Deno.test("noop test 2", () => { });
Deno.test("noop test 3", () => { });

clearTimeout(setTimeout(() => { }, 1000));

// Test will fail if remove this line:
await new Promise(resolve => setTimeout(resolve, 0));

// Deno starts running tests...
```

**Comment by magurotuna** [comment]
Seems like #7878 also has something to do with this issue

**Comment by qwerasd205** [comment]
> Seems like #7878 also has something to do with this issue

Yeah, it does seem to be caused by the same thing, an unrelated async op completing during a test causing the diffs to vary.

**Comment by KyleJune** [comment]
The reason awaiting another timeout resolves the issue is that it defers running the tests until next event loop turn. That way the timeouts/intervals cleared can be removed from the resource table. Adding a delay inside assertOps was done to work around a similar issue. Based on that fix, I think if deno were to add a delay before starting the test, it would resolve this issue.

https://github.com/denoland/deno/issues/4591

**Comment by bartlomieju** [comment]
@caspervonb could you take a look at this one?

**Comment by caspervonb** [comment]
> @caspervonb could you take a look at this one?

On it

**Comment by bartlomieju** [comment]
Fixed in v1.10.0
